From f56bf9b26f903b7d8b07bb0b4b7baf4dcb5562e0 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Tue, 8 Oct 2024 17:09:12 +0200
Subject: [PATCH 1322/1695] dt-bindings: remove CLK_CFG/CMD_CLK support for
 st,kerclk on STM32MP2

Remove the CMD_CLK support and CLK_CFG macro for st,kerclk,
because it is unused and not required in Binding (see OP-TEE file
documentation/devicetree/bindings/clock/st,stm32mp25-rcc.yaml).

The CLK_CMD command was defined for STM32MP1 SoC but is it no more
required for STM32MP2 in clock tree definition and in the RCC clock driver
(see stm32_clk_configure function and stm32_clk_kernel_configure
functions).

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 9a2ae64c076854f5a77c60f629a2c28fd9cd746c dt-bindings: remove CLK_CFG/CMD_CLK support for st,kerclk on STM32MP2]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I72475c50075b139f70945adc2bf8b1d81b0328e8
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/409883
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 .../dt-bindings/clock/stm32mp21-clksrc.h      | 27 -------------------
 .../dt-bindings/clock/stm32mp25-clksrc.h      | 27 -------------------
 2 files changed, 54 deletions(-)

diff --git a/core/include/dt-bindings/clock/stm32mp21-clksrc.h b/core/include/dt-bindings/clock/stm32mp21-clksrc.h
index 169602be9..0f837215e 100644
--- a/core/include/dt-bindings/clock/stm32mp21-clksrc.h
+++ b/core/include/dt-bindings/clock/stm32mp21-clksrc.h
@@ -8,7 +8,6 @@
 
 #define CMD_DIV		0
 #define CMD_MUX		1
-#define CMD_CLK		2
 #define CMD_FLEXGEN	3
 #define CMD_OBS		4
 
@@ -30,32 +29,6 @@
 #define MUX_SEL_SHIFT	0
 #define MUX_SEL_MASK	0x0000000F
 
-/* CLK define */
-#define CLK_ON_MASK	(1 << 21)
-#define CLK_ON_SHIFT	21
-
-#define CLK_ID_MASK	GENMASK_32(20, 12)
-#define CLK_ID_SHIFT	12
-
-#define CLK_NO_DIV_MASK	(1 << 7)
-#define CLK_DIV_MASK	GENMASK_32(10, 5)
-#define CLK_DIV_SHIFT	5
-
-#define CLK_NO_SEL_MASK	(1 << 4)
-#define CLK_SEL_MASK	GENMASK_32(3, 0)
-#define CLK_SEL_SHIFT	0
-
-#define CLK_CFG(clk_id, sel, div, state)	((CMD_CLK << CMD_SHIFT) |\
-						 ((state) << CLK_ON_SHIFT) |\
-						 ((clk_id) << CLK_ID_SHIFT) |\
-						 ((div) << CLK_DIV_SHIFT) |\
-						 ((sel) << CLK_SEL_SHIFT))
-
-#define CLK_OFF		0
-#define CLK_ON		1
-#define CLK_NODIV	(1 << 6)
-#define CLK_NOMUX	(1 << 4)
-
 /* Flexgen define */
 #define FLEX_ID_SHIFT	20
 #define FLEX_SEL_SHIFT	16
diff --git a/core/include/dt-bindings/clock/stm32mp25-clksrc.h b/core/include/dt-bindings/clock/stm32mp25-clksrc.h
index a02733601..7746f92b8 100644
--- a/core/include/dt-bindings/clock/stm32mp25-clksrc.h
+++ b/core/include/dt-bindings/clock/stm32mp25-clksrc.h
@@ -8,7 +8,6 @@
 
 #define CMD_DIV		0
 #define CMD_MUX		1
-#define CMD_CLK		2
 #define CMD_FLEXGEN	3
 #define CMD_OBS		4
 
@@ -30,32 +29,6 @@
 #define MUX_SEL_SHIFT	0
 #define MUX_SEL_MASK	0x0000000F
 
-/* CLK define */
-#define CLK_ON_MASK	BIT(21)
-#define CLK_ON_SHIFT	21
-
-#define CLK_ID_MASK	GENMASK_32(20, 12)
-#define CLK_ID_SHIFT	12
-
-#define CLK_NO_DIV_MASK	0x0000080
-#define CLK_DIV_MASK	GENMASK_32(10, 5)
-#define CLK_DIV_SHIFT	5
-
-#define CLK_NO_SEL_MASK	0x00000010
-#define CLK_SEL_MASK	GENMASK_32(3, 0)
-#define CLK_SEL_SHIFT	0
-
-#define CLK_CFG(clk_id, sel, div, state)	((CMD_CLK << CMD_SHIFT) |\
-						 ((state) << CLK_ON_SHIFT) |\
-						 ((clk_id) << CLK_ID_SHIFT) |\
-						 ((div) << CLK_DIV_SHIFT) |\
-						 ((sel) << CLK_SEL_SHIFT))
-
-#define CLK_OFF		0
-#define CLK_ON		1
-#define CLK_NODIV	0x00000040
-#define CLK_NOMUX	0x00000010
-
 /* Flexgen define */
 #define FLEX_ID_SHIFT	20
 #define FLEX_SEL_SHIFT	16
-- 
2.47.2

